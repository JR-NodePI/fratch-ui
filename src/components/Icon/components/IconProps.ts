export const IconType = Object.freeze({
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
  ERROR: 'error',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
});

export type IconProps = {
  svg: React.ReactElement<JSX.IntrinsicElements['svg']>;
  className?: string;
  type?: (typeof IconType)[keyof typeof IconType];
};
