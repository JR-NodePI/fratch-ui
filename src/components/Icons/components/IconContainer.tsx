import { c } from '../../../helpers/classNameHelpers';
import { IconType } from './IconConstants';
import { IconContainerProps } from './IconContainerProps';

import styles from './IconContainer.module.css';

const IconContainer = ({
  children,
  className,
  type = IconType.PRIMARY,
}: IconContainerProps): JSX.Element =>
  children({
    iconClassName: c(styles.icon, styles[type], className),
  });

export default IconContainer;
