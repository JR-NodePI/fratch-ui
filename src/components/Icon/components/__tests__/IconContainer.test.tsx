import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import IconContainer from '../IconContainer';
import { type IconProps, IconType } from '../IconProps';

describe('Button.tsx', () => {
  const svg = <svg></svg>;

  const setup = (props: IconProps = { svg }) =>
    render(<IconContainer {...props} />);

  it('should render properly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  describe('should render with TYPE', () => {
    Object.values(IconType).forEach(type => {
      it(type, () => {
        const { container } = setup({ type, svg });
        expect(container).toMatchSnapshot();
      });
    });
  });

  it('should render with className', () => {
    const { container } = setup({
      className: 'mock_test_class_name',
      svg,
    });
    expect(container).toMatchSnapshot();
  });
});
