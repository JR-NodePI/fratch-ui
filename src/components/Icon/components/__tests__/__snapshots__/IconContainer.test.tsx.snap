// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button.tsx > should render properly 1`] = `
<div>
  <div
    class="_icon_295c81 _primary_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > error 1`] = `
<div>
  <div
    class="_icon_295c81 _error_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > info 1`] = `
<div>
  <div
    class="_icon_295c81 _info_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > primary 1`] = `
<div>
  <div
    class="_icon_295c81 _primary_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > secondary 1`] = `
<div>
  <div
    class="_icon_295c81 _secondary_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > success 1`] = `
<div>
  <div
    class="_icon_295c81 _success_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > tertiary 1`] = `
<div>
  <div
    class="_icon_295c81 _tertiary_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with TYPE > warning 1`] = `
<div>
  <div
    class="_icon_295c81 _warning_295c81"
  >
    <svg />
  </div>
</div>
`;

exports[`Button.tsx > should render with className 1`] = `
<div>
  <div
    class="_icon_295c81 _primary_295c81 mock_test_class_name"
  >
    <svg />
  </div>
</div>
`;
